layers.go

```diff
@@ -2386,8 +2386,25 @@ func (r *layerStore) ApplyDiffFromStagingDirectory(id, stagingDirectory string,
 	layer.UncompressedDigest = diffOutput.UncompressedDigest
 	layer.UncompressedSize = diffOutput.Size
 	layer.Metadata = diffOutput.Metadata
-	if err = r.saveFor(layer); err != nil {
-		return err
+	if diffOutput.TarSplit != nil {
```

Comment:
I think `len(diffOutput.TarSplit) != 0` is more idiomatic.  I don't think it's ever `nil` for zstd chunked.